src/data/finset/locally_finite.lean

```diff
@@ -162,18 +230,30 @@ by rw [←coe_eq_singleton, coe_Icc, set.Icc_eq_singleton_iff]
 section decidable_eq
 variables [decidable_eq α]
 
-lemma Icc_erase_left : (Icc a b).erase a = Ioc a b :=
-by rw [←coe_inj, coe_erase, coe_Icc, coe_Ioc, set.Icc_diff_left]
-
-lemma Icc_erase_right : (Icc a b).erase b = Ico a b :=
-by rw [←coe_inj, coe_erase, coe_Icc, coe_Ico, set.Icc_diff_right]
+@[simp] lemma Icc_erase_left : (Icc a b).erase a = Ioc a b := by simp [←coe_inj]
+@[simp] lemma Icc_erase_right : (Icc a b).erase b = Ico a b := by simp [←coe_inj]
+@[simp] lemma Ico_erase_left : (Ico a b).erase a = Ioo a b := by simp [←coe_inj]
+@[simp] lemma Ioc_erase_right : (Ioc a b).erase b = Ioo a b := by simp [←coe_inj]
+@[simp] lemma Icc_diff_both : Icc a b \ {a, b} = Ioo a b := by simp [←coe_inj]
 
 lemma Ico_insert_right (h : a ≤ b) : insert b (Ico a b) = Icc a b :=
 by rw [←coe_inj, coe_insert, coe_Icc, coe_Ico, set.insert_eq, set.union_comm, set.Ico_union_right h]
 
 lemma Ioo_insert_left (h : a < b) : insert a (Ioo a b) = Ico a b :=
 by rw [←coe_inj, coe_insert, coe_Ioo, coe_Ico, set.insert_eq, set.union_comm, set.Ioo_union_left h]
 
+lemma Ioo_insert_right (h : a < b) : insert b (Ioo a b) = Ioc a b :=
+by rw [←coe_inj, coe_insert, coe_Ioo, coe_Ioc, set.insert_eq, set.union_comm, set.Ioo_union_right h]
+
+lemma Ioc_union_left (h : a ≤ b) : insert a (Ioc a b) = Icc a b :=
+by rw [←coe_inj, coe_insert, coe_Ioc, coe_Icc, set.insert_eq, set.union_comm, set.Ioc_union_left h]
+
+@[simp] lemma Ico_diff_Ioo_same (h : a < b) : Ico a b \ Ioo a b = {a} := by simp [←coe_inj, h]
```

Comment:
This departs from the `set` API but is in line with `Ioo_subset_Icc_self` and eventually I want to make the change to the `set` API too.
```suggestion
@[simp] lemma Ico_diff_Ioo_self (h : a < b) : Ico a b \ Ioo a b = {a} := by simp [←coe_inj, h]
```

Reply:
@YaelDillies Are we still waiting on this change? Or is the API inconsistent because of this?